packages/circuit-ui/components/ImageInput/ImageInput.tsx

```diff
@@ -293,21 +326,71 @@ export const ImageInput = ({
     clearInputElement();
   };
 
+  const handlePaste = (event: ClipboardEvent) => {
+    const { files } = event.clipboardData;
+    handleChange(files);
+
+    if (inputRef.current && files) {
+      // An error is thrown when trying to assign anything but a FileList here.
+      // For security reasons, it's not possible to simulate a FileList object.
+      // That's why this code has to be disabled during testing.
+      if (process.env.NODE_ENV !== 'test') {
+        inputRef.current.files = files;
+      }
+    }
+  };
+
+  const handleDragging = (event: DragEvent) => {
+    event.preventDefault();
+    event.stopPropagation();
+    setDragging(true);
+  };
+
+  const handleDragLeave = (event: DragEvent) => {
+    event.preventDefault();
+    event.stopPropagation();
+    setDragging(false);
+  };
+
+  const handleDrop = (event: DragEvent) => {
+    handleDragLeave(event);
+    const files = event.dataTransfer?.files;
```

Comment:
Is it safe to use optional chaining here? 👀 I think we talked about this but forgot what we said

Reply:
Hm, good point. I've gotten so used to it by now. 

Replaced in 9b3da0812.

Reply:
Just noticed we have a couple more in the Popover, could you do it in this PR? Otherwise I'll make another one

Reply:
This PR targets `main` which doesn't contain the Popover yet.

Reply:
Ah 🙈 will do on `next`

Reply:
I can also do it, need to make a bugfix on the Popover anyway.